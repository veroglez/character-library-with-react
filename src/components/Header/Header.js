import { object, string, oneOfType } from 'prop-types';
import React from 'react';
import { Image } from '..';
import scss from './Header.module.scss';

const Header = ({ logo }) => (
  <header className={scss.header}>
    {logo && <Image styles={scss.logo} image={logo} />}
  </header>
);

Header.propTypes = {
  logo: oneOfType([object, string]),
};

Header.defaultProps = {
  logo: null,
};

export default Header;
