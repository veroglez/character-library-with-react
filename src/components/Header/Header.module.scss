@import "../../common/style.scss";

.header {
  padding: 10px;
  border-bottom: 1px solid map-get($color, light);

  .logo {
    height: 40px;
    width: 150px;
    background-size: cover;
  }
}