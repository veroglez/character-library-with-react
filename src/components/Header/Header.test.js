import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

const IMAGE = 'image.png';

describe('<Header>', () => {
  it('renders', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {logo}', () => {
    const tree = renderer.create(<Header logo={IMAGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
