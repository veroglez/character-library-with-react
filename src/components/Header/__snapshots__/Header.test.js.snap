// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> renders 1`] = `<header />`;

exports[`<Header> when {logo} 1`] = `
<header>
  <div
    className="undefined"
    style={
      Object {
        "backgroundImage": "url(image.png)",
      }
    }
  />
</header>
`;
